.github/workflows/publish.yml

```diff
@@ -0,0 +1,24 @@
+name: Publish new version
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
```

Comment:
Double-check: do we want to publish if unit tests fail?

If not, we could move this job to `.github/workflows/test.yml`, and add an additional `if` condition to check the branch name (e.g. https://github.com/Brightspace/d2l.dev/blob/09aded170d6492302936cab5b18e75e9fb62641b/.github/workflows/gateway.yml#L34)

Reply:
Do we want to test on master and the PR? Is that what other repos do?

Reply:
Typically other repos test on pull requests and release on `master`, skipping tests on `master`. In theory that could cause problems... but I think the alternative would mean combining both workflows into one and then opting out of different steps based on the branch. 🤷

Reply:
Theory: if "ensure branches are up-to-date" is not enabled, it's possible two different changes could have passing tests but fail when both have been merged. Testing on `master` will prevent these cases from happening.

> Do we want to test on master and the PR? Is that what other repos do?

Not sure if you meant "release" instead of "test", but every PR should get tested, and only `master` should get released. You can prevent the release steps from running on a PR with 


```yaml
name: Semantic Release
if: github.ref == 'refs/heads/master'
...
```

Reply:
Sorry my sentence was stated weirdly, but I was wondering if we wanted the test step to run on master as well as on PR. It seems like it's easy enough to just do that anyway, so I'll make that change.